### ISSUE TITLE
Design System: AvatarIcons

### ISSUE THREAD

**Issue by mica000** [blockstack/blockstack-app#4884]
This ticket is for updating the tokens in the app:

**We have three sizes:**
40px - Tokens, Profiles, Collectibles, etc... = Flag layout
32px - Tokens, Profiles, Collectibles, etc... = Btn with Avatar
16px - Flag layout

![Image](https://github.com/leather-wallet/extension/assets/119523541/1c8c316a-c6ad-419b-acf5-d1383eea0fa2)

**Missing in current implementation**
- Add transparent 1px border to AvatarIcon 32px (@fab, we will need the semantic color here so it works on light and dark mode)
- Add white 2px border to Token 16px

![Image](https://github.com/leather-wallet/extension/assets/119523541/f659d525-b485-42bd-ba23-b46f477611b1)

**Comment by mica000** [comment]
@kyranjamie curious on your thoughts around this component so we can set up the same way it will handle by implementation

**Comment by kyranjamie** [comment]
Nothing to note here. Pretty sure we'll implement these the same as icons, just with only two possible sizes.

**Comment by mica000** [comment]
@kyranjamie sorry, something happened and it didnt save the alteration I had done on the ticket. We actually have 3 sizes of Avatars

**Comment by fabric-8** [comment]
@mica000 How do you feel about changing the way we build this component? [Put together an alternative component proposal](https://www.figma.com/file/2MLHeIeL6XPVi3Tc2DfFCr/%E2%9D%96-Leather-%E2%80%93-Design-System?type=design&node-id=13104%3A69208&mode=design&t=MjIjd8u00zNouLoh-1) which I think is more in line with the way we're using it and also includes all sizes.

TLDR: Size (16x16, 24x,24x24, 32x32) as well as the type (token or icon) are both just props of the component.

https://github.com/leather-wallet/extension/assets/4112698/a21037dd-3666-4773-a544-5f501cbb57ca

**Comment by mica000** [comment]
@fabric-8 Yes!! seems to replicate better what Kyran mentioned above :) thanks Fab!!!

**Comment by mica000** [comment]
@kyranjamie ok to move forward?

**Comment by fbwoolf** [comment]
It seems like the video here has an impact on the Icons issue? I should do this work on that branch, so I'll start this as part of that work.

**Comment by mica000** [comment]
@fbwoolf Indeed. thanks!

**Comment by fbwoolf** [comment]
Making a comment here to document my questions in Figma regarding this issue.

1. Sizes differ b/w this issue and what is shown in Figma, so I need clarification on what sizes we want available and whether or not it is ok for the icons to scale in the 16x16 version? The icon used in the Figma example is 10x10. We currently only have svgs exported for icons that are 16x16 and 24x24.
2. The naming seems off to me if we want two `types` of avatar icons, should the name of this component actually just be `Avatar` so it can expand later to possibly just show an image? https://www.radix-ui.com/primitives/docs/components/avatar
3. Some of the `tokens` shown are not actually tokens, so curious what is the thinking there if the type is `token` (ex. Ordinals, Stamps, the one showing just an emoji)?
4. Why are some square?

**Comment by fabric-8** [comment]
Responding here as well as it might be easier to consume vs. following through the various threads on figma:

> Sizes differ b/w this issue and what is shown in Figma, so I need clarification on what sizes we want available and whether or not it is ok for the icons to scale in the 16x16 version? The icon used in the Figma example is 10x10. We currently only have svgs exported for icons that are 16x16 and 24x24.

Yes, let's scale the 16x16 icons for now here. To my knowledge we're currently only using those icons only for activity which will be redesigned in the context of the web app. When testing how those come out, 10x10 looked best IMO.

> The naming seems off to me if we want two types of avatar icons, should the name of this component actually just be Avatar so it can expand later to possibly just show an image? https://www.radix-ui.com/primitives/docs/components/avatar

Agree — The avatar component can basically consume either icons or images (PNGs/SVGs). Question regarding emojis below

> Some of the tokens shown are not actually tokens, so curious what is the thinking there if the type is token (ex. Ordinals, Stamps, the one showing just an emoji)?

Except emojis, I think ultimately we're indeed talking images here (mostly logos as a specific kind of image). Not sure about the emoji one: That's something that has been in Figma, but not sure if it's actually used in prod — If yes, that might be another type?

> Why are some square?

That's in case of collectibles where we basically display a scaled down version of the image as a thumbnail / preview
